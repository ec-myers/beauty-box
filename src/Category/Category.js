import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import './Category.scss';

export const Category = ({ isLoading }) => {
  return (
    <>
    <div className='hero-div'>
      <img src={require('../Images/hero_img_left.jpg')} alt='woman smiling at the camera'/>
      <div className='shop-div'>
        <h3>EXPLORE</h3>
        <h3>BEAUTY</h3>
          <Link to='/shopall'><button>SHOP ALL</button></Link>
      </div>
        <img src={require('../Images/hero_image_right.jpg')} alt='perfume on a blanket'></img>
    </div>
    {isLoading && <img className='loading-img' src={require('../Images/loading.gif')} alt='loading' />}
    {!isLoading && <div className='categories-div'>
      <Link to={ '/products/foundations'}> 
        <div className='foundation-div'>
          <h4>FOUNDATION</h4>
        </div> 
      </Link>
      <Link to='/products/blushes'>
        <div className='blush-div'>
          <h4>BLUSH</h4>
        </div>
      </Link>
      <Link to='/products/mascaras'>
        <div className='mascara-div'>
          <h4>MASCARA</h4>
        </div>
      </Link>
      <Link to='/products/eyeshadows'>
        <div className='eyeshadow-div'>
          <h4>EYESHADOW</h4>
        </div>
      </Link>
      <Link to='/products/lipsticks'>
        <div className='lipstick-div'>
          <h4>LIPSTICK</h4>
        </div>
      </Link>
    </div>}
    </>
  )
}

export const mapStateToProps = ({ isLoading }) => ({
  isLoading
});

export default connect(mapStateToProps)(Category);