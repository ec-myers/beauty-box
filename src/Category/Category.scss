.hero-div {
  display: flex;
  flex-direction: row;
  height: 500px;
  justify-content: space-evenly
}