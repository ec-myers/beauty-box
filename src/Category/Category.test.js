import React from 'react';
import { shallow } from 'enzyme';
import { Category, mapStateToProps } from './Category';

describe('Category', () => {
  it('should match the snapshot', () => {
    let wrapper = shallow(<Category isLoading={false} />)
    
    expect(wrapper).toMatchSnapshot();
  });

});
