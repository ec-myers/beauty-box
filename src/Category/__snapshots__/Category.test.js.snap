// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category should match the snapshot 1`] = `
<Fragment>
  <div
    className="hero-div"
  >
    <img
      alt="woman smiling at the camera"
      src="hero_img_left.jpg"
    />
    <div
      className="shop-div"
    >
      <h3>
        EXPLORE
      </h3>
      <h3>
        BEAUTY
      </h3>
      <Link
        to="/shopall"
      >
        <button>
          SHOP ALL
        </button>
      </Link>
    </div>
    <img
      alt="perfume on a blanket"
      src="hero_image_right.jpg"
    />
  </div>
  <div
    className="categories-div"
  >
    <Link
      to="/products/foundations"
    >
      <div
        className="foundation-div"
      >
        <h4>
          FOUNDATION
        </h4>
      </div>
    </Link>
    <Link
      to="/products/blushes"
    >
      <div
        className="blush-div"
      >
        <h4>
          BLUSH
        </h4>
      </div>
    </Link>
    <Link
      to="/products/mascaras"
    >
      <div
        className="mascara-div"
      >
        <h4>
          MASCARA
        </h4>
      </div>
    </Link>
    <Link
      to="/products/eyeshadows"
    >
      <div
        className="eyeshadow-div"
      >
        <h4>
          EYESHADOW
        </h4>
      </div>
    </Link>
    <Link
      to="/products/lipsticks"
    >
      <div
        className="lipstick-div"
      >
        <h4>
          LIPSTICK
        </h4>
      </div>
    </Link>
  </div>
</Fragment>
`;
